kotlin_money/src/main/java/com/github/paylike/kotlin_money/Money.kt

```diff
@@ -0,0 +1,32 @@
+package com.github.paylike.kotlin_money
+
+import com.github.paylike.kotlin_currencies.PaylikeCurrency
+
+object Money {
+    private const val maxInt: Long = 9007199254740991L
+
+    private fun isInSafeRange(n: Double): Boolean = (n <= maxInt && n >= -maxInt)
+
+    fun fromDouble(currency: PaylikeCurrency, n: Double): PaymentAmount {
+        if (!n.isFinite()) {
+            throw Exception("Non finite number $n")
```

Comment:
This could also be `UnsafeNumberException`

Reply:
You are right, but in this case we should change the dart version.